### PR TITLE
kv/client: add incremental scan region count limit

### PR DESCRIPTION
**Pull request by amyangfei**
### What problem does this PR solve? 

cherry-pick #1899 into release-4.0, for test only

### What is changed and how it works?

cherry-pick #1899 into release-4.0, for test only

### Check List 

Tests 

 - Unit test
 - Integration test

### Release note

- Add concurrency limit to the region incremental scan in kv client.

### CODE REVIEW

**Review by overvenus — APPROVED** [review]
LGTM

**Review by liuzix — APPROVED** [review]

File: cdc/kv/client.go [hunk 1]
```diff
@@ -68,6 +68,9 @@ const (
 	// failed region will be reloaded via `BatchLoadRegionsWithKeyRange` API. So we
 	// don't need to force reload region any more.
 	regionScheduleReload = false
+
+	// defines the scan region limit for each table
+	regionScanLimitPerTable = 40
```
Comment (overvenus) [inline cdc/kv/client.go:73]:
```suggestion
	regionScanLimitPerTable = 6
```

Comment (amyangfei) [reply cdc/kv/client.go:73]: lgtm

File: cdc/kv/client_test.go [hunk 2]
```diff
@@ -3138,7 +3138,9 @@ func (s *etcdSuite) TestConcurrentProcessRangeRequest(c *check.C) {
 	lockresolver := txnutil.NewLockerResolver(kvStorage.(tikv.Storage))
 	isPullInit := &mockPullerInit{}
 	cdcClient := NewCDCClient(ctx, pdClient, kvStorage.(tikv.Storage), &security.Credential{})
-	eventCh := make(chan *model.RegionFeedEvent, 10)
+	// The buffer size of event channel must be large enough because in the test
+	// case we send events first, and then retrive all events from this channel.
+	eventCh := make(chan *model.RegionFeedEvent, 100)
```
Comment (amyangfei) [inline cdc/kv/client_test.go:3143]: Note the fix for this bug should be picked to master @amyangfei

**Pull request merged**